diff --git a/lib/rules/max-lines-per-function.js b/lib/rules/max-lines-per-function.js
index 80f93ea4ee57497904a51e7d6031f0355d8b2981..0bb3914b981d8d840c84099d5335e3dac6cf361a 100644
--- a/lib/rules/max-lines-per-function.js
+++ b/lib/rules/max-lines-per-function.js
@@ -279,8 +279,8 @@ module.exports = {
             const baseFunctionLineNum = 2
             lineCount = lineCount - argumentsLineNum - baseFunctionLineNum;
 
-            const isOverLength = lineCount > maxLines;
-            const isError = isOverLength && !isReactEle(node);
+            const maxLinesToCheck = (maxLines >= 150 || !isReactEle(node)) ? maxLines : 150;
+            const isError = lineCount > maxLinesToCheck;
 
             if (isError) {
                 const functionName = getFunctionNameWithKind(node);
@@ -289,7 +289,7 @@ module.exports = {
                 context.report({
                     node,
                     messageId: "exceed",
-                    data: { name, lineCount, maxLines }
+                    data: { name, lineCount, maxLines: maxLinesToCheck }
                 });
             }
         }
